src/daft-dsl/src/functions/python/mod.rs

```diff
@@ -29,11 +29,53 @@ impl PythonUDF {
     }
 }
 
+/// A wrapper around PyObject that is safe to use even when the Python feature flag isn't turned on
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct RuntimePyObject {
+    #[cfg(feature = "python")]
+    obj: crate::pyobj_serde::PyObjectWrapper,
+}
+
+impl RuntimePyObject {
+    #[cfg(feature = "test-utils")]
+    pub fn new_testing_none() -> Self {
+        #[cfg(feature = "python")]
+        {
+            let none_value = pyo3::Python::with_gil(|py| py.None());
+            Self {
+                obj: crate::pyobj_serde::PyObjectWrapper(none_value),
+            }
+        }
+        #[cfg(not(feature = "python"))]
+        {
+            Self {}
+        }
+    }
+
+    #[cfg(feature = "python")]
+    pub fn new(value: pyo3::PyObject) -> Self {
+        Self {
+            obj: crate::pyobj_serde::PyObjectWrapper(value),
+        }
+    }
+
+    #[cfg(feature = "python")]
+    pub fn unwrap(&self) -> &pyo3::PyObject {
```

Comment:
you can make this unwrap take in `self` and move out `pyo3::PyObject`. 

For the reference you can impl AsRef
https://doc.rust-lang.org/stable/std/convert/trait.AsRef.html

Reply:
Good call -- changed!